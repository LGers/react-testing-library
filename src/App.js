import React, { useState } from "react"
import logo from './logo.svg';
import './App.css';

const Search = ({value, onChange, children}) => (
  <div>
    <label htmlFor='search'>{children}</label>
    <input id="search" type="text" value={value} onChange={onchange} placeholder='search text...'></input>
  </div>
)

const App = () => {

  const [search, setSearch] = useState("")

  const handleChange = ({ target }) => {
    setSearch(target.value)
  }

  return (
    <div>
      <img src='' alt='search image'/>
      <Search value={search} onChange={handleChange}>
        Search:
      </Search>
      <p>Searches for {search ? search : "..."}</p>
    </div>
    // <div className="App">
    //   <header className="App-header">
    //     <img src={logo} className="App-logo" alt="logo" />
    //     <p>
    //       Edit <code>src/App.js</code> and save to reload.
    //     </p>
    //     <a
    //       className="App-link"
    //       href="https://reactjs.org"
    //       target="_blank"
    //       rel="noopener noreferrer"
    //     >
    //       Learn React testing library
    //     </a>
    //   </header>
    // </div>
  )
}

export default App;
