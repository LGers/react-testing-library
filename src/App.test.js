import { render, screen } from '@testing-library/react';
import App from './App';

/*
Search variants:
  getBy:                    queryby:                    findBy:
- getByText               - queryByText               - findByText
- getByRole               - queryByRole               - findByRole
- getByLabelText          - queryByLabelText          - findByLabelText
- getByPlaceholderText    - queryByPlaceholderText    - findByPlaceholderText
- getByAltText            - queryByAltText            - findByAltText
- getByDisplayValue       - queryByDisplayValue       - findByDisplayValue
- getAllBy                - queryAllBy                - findAllBy
*/

/*
Assertive Functions:
- toBeDisabled            - toBeEnabled               - toBeEmpty
- toBeEmptyDOMElement     - toBeInTheDocument         - toBeInvalid
- toBeRequired            - toBeValid                 - toBeVisible
- toContainElement        - toContainHTML             - toHaveAttribute
- toHaveClass             - toHaveFocus               - toHaveFormValues
- toHaveStyle             - toHaveTextContent         - toHaveValue
- toHaveDisplayValue      - toBeChecked               - toBePartiallyChecked
- toHaveDescription
*/

//lesson-4

describe("App", () => {
  it("renders App component", async () => {
    render(<App />);
    // expect(screen.queryByText(/Searches for React/i)).toBeNull();
    expect(screen.queryByText(/Logged in as/i)).toBeNull();
    screen.debug();
    expect(await screen.findByText(/Logged in as/i)).toBeInTheDocument();
    screen.debug();
    expect(screen.getByAltText(/image/i)).toHaveClass('image');
    expect(screen.getByLabelText(/search/i)).not.toBeRequired();
    expect(screen.getByLabelText(/search/i)).toBeEmpty();
    expect(screen.getByLabelText(/search/i)).toHaveAttribute("id");
  });
});
