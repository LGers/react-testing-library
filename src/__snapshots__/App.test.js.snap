// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link2 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="search image"
      src=""
    />
    <div>
      <label
        for="search"
      >
        Search:
      </label>
      <input
        id="search"
        placeholder="search text..."
        type="text"
        value=""
      />
    </div>
    <p>
      Searches for ...
    </p>
  </div>
</DocumentFragment>
`;
